torch_geometric/sampler/utils.py

```diff
@@ -0,0 +1,106 @@
+from typing import Any, Dict, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.storage import EdgeStorage
+from torch_geometric.typing import EdgeType, OptTensor
+
+# Since C++ cannot take dictionaries with tuples as key as input, edge type
+# triplets need to be converted into single strings. This is done by adding
+# EDGE_TYPE_SPLIT_STR between values in the edge type tuple:
+EDGE_TYPE_SPLIT_STR = "__"
+
+# Edge Type Conversion ########################################################
+
+
+def edge_type_to_str(edge_type: Union[EdgeType, str]) -> str:
+    if isinstance(edge_type, str):
+        return edge_type
+    return EDGE_TYPE_SPLIT_STR.join(edge_type)
+
+
+def edge_type_to_str_dict(
+        edge_type_dict: Dict[EdgeType, Any]) -> Dict[str, Any]:
+    return {edge_type_to_str(k): v for k, v in edge_type_dict.items()}
+
+
+def edge_type_from_str(edge_type: Union[EdgeType, str]) -> EdgeType:
```

Comment:
Do we need this function? This is generally dangerous since it means we do not support any types with `__` in its name.

Reply:
The reason this is included is that it allows samplers to speak purely with edge types, instead of with the string edge types (currently, loaders filter based on `edge_type_to_str`). I agree that it is dangerous for that reason; the alternative would be to define a custom `to_str` and `from_str` here that are PyG-specific, but that can lead to weird debugging outputs, or we can speak through string edge types (but this leads to using `edge_type_to_str` in lots of places, since GraphStore currently speaks through edge types). Wdyt?

Reply:
I think a workaround is to maintain this mapping explicitely, see https://github.com/pyg-team/pyg-lib/blob/master/pyg_lib/sampler/__init__.py#L86-L87

Reply:
Sure, that works too.